### ISSUE TITLE
Verification for surface O3 and PM2.5

### ISSUE THREAD

**Issue by weiwilliam** [weiwilliam/Wx-AQ_LuLab#57]
Verify the chemistry forecasts at selected sites for the last 7/14/30 days.
Verification time window: previous day 12z to current day 12z (i.e., 12- 36 hours forecasts from previous 00Z run)

- [ ] pull AirNow data
- [ ] post-process wrfout to generate files at selected stations.
- [ ] create timeseries plot
- [ ] against NOAA models

**Comment by SarahLu-NOAA** [comment]
Kara suggests that we pull the data directly from the airnow API.
Here is the info from Kara:
Here is a sample API: https://www.airnowapi.org/aq/data/?startDate=2022-12-27T18&endDate=2022-12-27T19&parameters=OZONE,PM25&BBOX=-79.998703,40.436229,-71.693039,45.419415&dataType=B&format=text/csv&verbose=1&monitorType=2&includerawconcentrations=1&API_KEY=2E960604-B2BF-42B5-83F6-B35C1B09DD1B

The BBOX should cover all of NY, and you can see the start and end dates can be adjusted per your specifications.
Not that the API_KEY used here is mine, under my account. I am assuming you have an account so you could just replace that KEY, though its probably not a huge deal either way.  The data returned from the API fetch is text/csv

**Comment by SarahLu-NOAA** [comment]
@clin-nick @Hui4ying3LUO2 
have you used the EPA Airnow API to pull the data before?   We will need to set up daily data feed to verify the WRF-Chem output.
Hui-ying, if you can set up this quickly before the end of the year, that could be helpful.  If you can't, no worries and do let Chin-An know.    Thanks anyway.

**Comment by weiwilliam** [comment]
The cronjob of pulling AirNow data is under testing now.
The first AirNow data should be available in /network/asrc/scratch/lulab/sw651133/nomads_dev soon.

**Comment by SarahLu-NOAA** [comment]
Thanks @weiwilliam  for the update.
I discussed AQF data with Jeff at AMS.   Will follow up on this with Ho-Chun and Jiangping.

**Comment by clin-nick** [comment]
It looks like airnow data in /network/asrc/scratch/lulab/sw651133/nomads/chem/airnow/ did not be downloaded successfully for the past five days.

**Comment by weiwilliam** [comment]
To strengthen AirNow security, the AirNow API will force HTTPS redirects as of Friday, Feb 3rd, 2023. Please use HTTPS to access URLs for the AirNow API web services.

If you have any questions, please contact dmc@airnowtech.org.

Thank you,
EPA and AirNow Data Management Center.  
.... :expressionless:

**Comment by weiwilliam** [comment]
I see something wrong, let me check.

**Comment by weiwilliam** [comment]
@clin-nick it's fixed, I'm pulling data after Feb 3.

**Comment by clin-nick** [comment]
I just found the date in the filename and csv file do not match.
e.g., in airnow_20230203.csv,  
                 time
0       2023-02-02T00:00
1       2023-02-02T00:00
2       2023-02-02T00:00
3       2023-02-02T00:00
4       2023-02-02T00:00

**Comment by weiwilliam** [comment]
Yeah! Thanks for catching up, I forgot to update the name of outputfile when refreshing the prior day's data.

**Comment by clin-nick** [comment]
It looks good now. Thanks!

**Comment by SarahLu-NOAA** [comment]
ACOM's forecast page
https://www.acom.ucar.edu/firex-aq/forecast.shtml

**Comment by SarahLu-NOAA** [comment]
@clin-nick 
Add WRF-Chem vs AIRNOW for 00z, 06z, 12z, and 18z

**Comment by weiwilliam** [comment]
Link #83 to here
